src/features/hotspots/setup/HotspotSetupExternalScreen.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useCallback, useMemo } from 'react'
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native'
+import { useTranslation } from 'react-i18next'
+import QrIcon from '@assets/images/qr.svg'
+import LinkIcon from '@assets/images/webLink.svg'
+import { BarCodeScanner, BarCodeScannerResult } from 'expo-barcode-scanner'
+import { Camera } from 'expo-camera'
+import { useAsync } from 'react-async-hook'
+import { useDebouncedCallback } from 'use-debounce/lib'
+import Toast from 'react-native-simple-toast'
+import { StyleSheet, Linking, ScrollView } from 'react-native'
+import Clipboard from '@react-native-community/clipboard'
+import { TouchableOpacity } from 'react-native-gesture-handler'
+import BackScreen from '../../../components/BackScreen'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import { HotspotSetupStackParamList } from './hotspotSetupTypes'
+import { useColors, useBorderRadii } from '../../../theme/themeHooks'
+import { getSecureItem } from '../../../utils/secureAccount'
+import { useAppLinkContext } from '../../../providers/AppLinkProvider'
+import useHaptic from '../../../utils/useHaptic'
+import { RootNavigationProp } from '../../../navigation/main/tabTypes'
+import { HotspotMakerModels } from '../../../makers'
+
+type Route = RouteProp<HotspotSetupStackParamList, 'HotspotSetupExternalScreen'>
+
+const HotspotSetupExternalScreen = () => {
+  const { t } = useTranslation()
+  const { params } = useRoute<Route>()
+  const colors = useColors()
+  const { xl } = useBorderRadii()
+  const { result: address } = useAsync(getSecureItem, ['address'])
+  const { handleBarCode } = useAppLinkContext()
+  const { triggerNotification } = useHaptic()
+  const navigation = useNavigation<RootNavigationProp>()
+
+  const isQr = useMemo(
+    () => HotspotMakerModels[params.hotspotType].onboardType === 'QR',
+    [params.hotspotType],
+  )
+
+  const handleClose = useCallback(() => navigation.navigate('MainTabs'), [
+    navigation,
+  ])
+
+  const handleBarCodeScanned = useDebouncedCallback(
```

Comment:
`use-debounce` seems nice, I have a place where I should consider using that in explorer